# `no-for-in`

Disallows "for x in y" statements.

## Rule Details

```ts
// Bad

// Good
```

## Options and Defaults

```json
{
  "rules": {
    "isaacscript/no-for-in": "error"
  }
}
```

This rule is not configurable.

## Resources

- [Rule source](../../src/rules/no-for-in.ts)
- [Test source](../../tests/rules/no-for-in.test.ts)
